pkg/upgrade/upgrade.go

```diff
@@ -326,17 +334,33 @@ func (u *Upgrader) getFlist(repo, name string) (meta.Walker, error) {
 // install from a single flist.
 func (u *Upgrader) install(repo, name string) error {
 	log.Info().Str("repo", repo).Str("name", name).Msg("start installing package")
+	inMemory := false
 
 	store, err := u.getFlist(repo, name)
 	if err != nil {
-		return errors.Wrapf(err, "failed to process flist: %s/%s", repo, name)
+		// try in memory
```

Comment:
I am wondering if handling ALL errors and do an in memory update is the right thing to do. I mean we already know if we have a cache problem (flags from storage are set if there is no cache and or cache is read only)

I think a better approach was to basically you check what is the status of system announced by storage, and then decide if u gonna do in-memory or normal update

Reply:
I mean what if this is just a network issue, and nothing wrong with storage?

This also mean that the decision to use a in-memory update, or not has to be decided at the highest level (in the Run implementation to be exact) when making the call to update.

Reply:
This also means, instead of passing around an optional `inMemory` flag, you can instead pass a `Cache` interface object (or any other name) that returns different passes where the files are stored (this can be point to persisted cache, or /tmp)

Why this is better than using a inMemory flag, simply because I don't have to do `if` every where since i can simply do `cache.Flist()` or `cache.File()` which returns the proper (created) paths.